src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Util.daml

```diff
@@ -422,12 +433,13 @@ calculateNotionalAndFixings p s
       Some fxRateId -> if fxAdjRequired then notionalConst * fxRateObs else notionalConst
         where fxRateObs = Observe fxRateId
     paymentPeriodData = map
-      (\c -> (
-        c.calculationPeriod,
-        c.rateFixingDate,
-        Const (calcPeriodDcf s.calculationPeriodAmount.calculation.dayCountFraction c.calculationPeriod
-        useAdjustedDatesForDcf periodicSchedule.terminationDate periodicSchedule.frequency)
-      ))
+      (\c -> CalculationPeriodData with
```

Comment:
Thanks, the additional record type helps a lot for readability!

One last question: in Fpml, is there a difference between `calculation period` and `payment period`? I seem to remember so, but I might be mistaken.

Reason I'm asking is that the record type is called `CalculationPeriodData` (hence referring to a Calc period) but it is bound to a variable called `Payment Period data`

Reply:
Yes, rates are determined in the calculation period and paid in the payment period. Often, they match 1-1, but in case of compounding there are e.g. 3 calc periods for each payment period. With the new logic, I iterate over the payment periods (and compound the calculation periods belonging to each payment period).